src/languages/swift.js

```diff
@@ -283,6 +288,143 @@ export default function(hljs) {
   };
   TYPE.contains.push(GENERIC_ARGUMENTS);
 
+  // https://docs.swift.org/swift-book/ReferenceManual/Expressions.html#ID552
+  // Prevents element names from being highlighted as keywords.
+  const TUPLE_ELEMENT_NAME = {
+    begin: lookahead(concat(Swift.identifier, /\s*:/)),
+    end: lookahead(/:/),
+    keywords: "_",
+    relevance: 0
+  };
+  // Matches tuples as well as the parameter list of a function type.
+  const TUPLE = {
+    begin: /\(/,
+    end: /\)/,
+    relevance: 0,
+    keywords: KEYWORDS,
+    contains: [
+      'self',
+      TUPLE_ELEMENT_NAME,
+      ...COMMENTS,
+      ...KEYWORD_MODES,
+      ...BUILT_INS,
+      ...OPERATORS,
+      NUMBER,
+      STRING,
+      ...IDENTIFIERS,
+      ...ATTRIBUTES,
+      TYPE
+    ]
+  };
+
+  // https://docs.swift.org/swift-book/ReferenceManual/Declarations.html#ID362
+  // Matches both the keyword func and the function title.
+  // Grouping these lets us differentiate between the operator function <
+  // and the start of the generic parameter clause (also <).
+  const FUNCTION_TITLE = {
+    begin: lookahead(/\bfunc\b/),
+    relevance: 0,
+    contains: [
+      {
+        className: 'keyword',
+        begin: /\bfunc\b/
+      },
+      {
+        begin: /\s+/,
+        relevance: 0
+      },
+      {
+        className: 'title',
+        begin: either(QUOTED_IDENTIFIER.begin, Swift.identifier, Swift.operator),
+        // Required to make sure the opening < of the generic parameter clause
+        // isn't parsed as a second title.
+        endsParent: true,
+        relevance: 0
+      }
+    ]
+  };
+  const GENERIC_PARAMETERS = {
+    begin: /\s*</,
+    end: />/,
+    contains: [
+      ...COMMENTS,
+      TYPE
+    ]
+  };
+  const FUNCTION_PARAMETER_NAME = {
+    begin: either(
+      lookahead(concat(Swift.identifier, /\s*:/)),
+      lookahead(concat(Swift.identifier, /\s+/, Swift.identifier, /\s*:/))
+    ),
+    end: lookahead(/:/),
+    relevance: 0,
+    contains: [
+      {
+        className: 'keyword',
+        begin: /\b_\b/
+      },
+      {
+        className: 'params',
+        begin: Swift.identifier
+      }
+    ]
+  };
+  const FUNCTION_PARAMETERS = {
+    begin: /\s*\(/,
```

Comment:
Ditto.  This rule probably belongs inside FUNCTION contains if the goal is to allow space.

Reply:
Yes, but only in a few places. I thought it was neater to allow them only in specific places, and not everywhere (which would be the result of a contained rule for spaces, as would trailing spaces.

Reply:
Repeating yourself vs doing it in a single place is usually the wrong approach.  Instead of every child including `\s*` it should be in a single place.  And it's not "everywhere". There are 3 rules in FUNCTION and 2 of them include this... and it won't harm the 3rd in any case.

Reply:
Fixed (see above).